api-tests/platform/targets/tgt_dev_apis_stdc/target.cmake

```diff
@@ -15,6 +15,63 @@
 # * limitations under the License.
 #**/
 
+###############################################################################
+# FUNCTION: _create_psa_stdc_exe()
+#  Function for generating PSA stdc test binaries linking with libraries
+#  external to the psa-arch-tests project. This function requires
+#  PSA_STORAGE_LIB_FILENAME to be specificed on the cmake command line,
+#  where the symbol is defined as the full path to the external PSA storage
+#  library to test. e.g.
+#    cmake ... -DPSA_STORAGE_LIB_FILENAME=/wdir/usr/lib/libpsastorage.so
+#  If the function is being used to generate a test binary for testing 
+#  the mbed-crypto library then the function requires 
+#  PSA_CRYPTO_LIB_FILENAME to be specificed on the cmake command line,
+#  where the symbol is defined as the full path to the external PSA crypto
+#  library to test. e.g.
+#    cmake ... -DPSA_CRYPTO_LIB_FILENAME=/wdir/mbed-crypto/library/    \
+#                                                               libmbedcrypto.a
+# ARGUMENTS:
+#   _exe_name     Name of the test binary to generate.
+#   _api_dir      PSA API directory name e.g. crypto, 
+#                 internal_trusted_storage or protected_storage.
+###############################################################################
+function(_create_psa_stdc_exe _exe_name _api_dir)
+
+	# Create the PSA test binary.
+	set(EXE_NAME ${_exe_name})
+
+	if(DEFINED PSA_STORAGE_LIB_FILENAME)
+		# Define PSA_STORAGE_LIB_NAME to be the name of the PSA Storage library to be tested.
+		get_filename_component(PSA_STORAGE_LIB_NAME ${PSA_STORAGE_LIB_FILENAME} NAME)
```

Comment:
This doesn't work for libraries names with long extensions (.a.b.c  eg: libpsastorage.so.1.0.0)
My suggestion is to remove this line and add ${PSA_STORAGE_LIB_FILENAME} at line 68
This will allow the library names to be independent.

Same comments for crypto.

Reply:
i dont understand this comment. I perform my testing using this Makefile:

https://github.com/ARMmbed/psa_trusted_storage_linux/blob/sdh_iotmbl_2565_psa_storage_building_test_binaries/test/Makefile

(The above file isnt on master because I need this PR merged first). At lines 95-97 in the above file you have these line:

```
# PSA_STORAGE_LIB_FILENAME is the location of the installed
# psa_trusted_storage_linux library
PSA_STORAGE_LIB_FILENAME=$(TOPDIR)/$(PSA_STORAGE_DIR)/$(PREFIX)/lib/libpsastorage.so
```
If you change line 97 to the following:

```
PSA_STORAGE_LIB_FILENAME=$(TOPDIR)/$(PSA_STORAGE_DIR)/$(PREFIX)/lib/libpsastorage.so.1.0.0
```

Then everything works fine with the version qualified libpsstorage.so.1.0.0, which is contrary to your findings i.e. you say it doesn't work. If you think about it, why wouldn't this work? Its just a library with a different name. This suggests there is some other cause for the problem you've encountered other than the name of the library.


P